storage_broker/src/bin/storage_broker.rs

```diff
@@ -734,13 +795,36 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         }
         .await;
 
+        let tls_acceptor = tls_acceptor.clone();
+
         tokio::task::spawn(async move {
-            let res = builder
-                .serve_connection(TokioIo::new(stream), service_fn_)
-                .await;
+            if is_https {
+                let tls_acceptor =
+                    tls_acceptor.expect("tls_acceptor is set together with https_listener");
+
+                let tls_stream = match tls_acceptor.accept(tcp_stream).await {
+                    Ok(tls_stream) => tls_stream,
+                    Err(e) => {
+                        info!("error accepting TLS connection from {addr}: {e}");
+                        return;
+                    }
+                };
+
+                let res = builder
+                    .serve_connection(TokioIo::new(tls_stream), service_fn_)
+                    .await;
 
-            if let Err(e) = res {
-                info!("error serving connection from {addr}: {e}");
+                if let Err(e) = res {
+                    info!("error serving HTTPS connection from {addr}: {e}");
+                }
+            } else {
+                let res = builder
+                    .serve_connection(TokioIo::new(tcp_stream), service_fn_)
+                    .await;
+
+                if let Err(e) = res {
+                    info!("error serving HTTP connection from {addr}: {e}");
```

Comment:
nit: instead of having two different strings, include `is_https` as a structured logging field